File: click.py
import pyautogui
import time
import random

# 自动获取屏幕分辨率
screen_width, screen_height = pyautogui.size()

# 计算屏幕中心
center_x, center_y = screen_width // 2, screen_height // 2

# 定义点击的范围，以中心点为基准，每个方向偏移50像素
range_x = 50
range_y = 50

# 点击次数
clicks = 100000000000

try:
    for _ in range(clicks):
        # 生成在中心区域的随机坐标
        x = random.randint(center_x - range_x, center_x + range_x)
        y = random.randint(center_y - range_y, center_y + range_y)

        # 模拟鼠标移动到目标位置，增加一些随机性
        move_duration = random.uniform(0.1, 0.3)  # 随机移动时间
        pyautogui.moveTo(x, y, duration=move_duration)

        # 模拟鼠标悬停效果，增加一点随机延时
        hover_time = random.uniform(0.1, 0.3)
        time.sleep(hover_time)

        # 模拟鼠标按下和抬起
        pyautogui.mouseDown()
        time.sleep(random.uniform(0.05, 0.1))  # 短暂的按下状态
        pyautogui.mouseUp()

        # 模拟点击后的微小调整或抖动，增加真实感
        pyautogui.moveRel(random.randint(-3, 3), random.randint(-3, 3), duration=0.05)

        # 生成a到b随机浮点数作为间隔时间
        pause_between_clicks = random.uniform(0.01, 0.1)

        # 暂停
        time.sleep(pause_between_clicks)

except KeyboardInterrupt:
    print("\n脚本已被中断。")

print("随机点击完成！")

File: click.js
// 定义停止时间戳（例如设置为当前时间+30秒）
let stopTime = Date.now() + 86400 * 1000;

function randomCenterClick() {
    // 获取当前时间
    const currentTime = Date.now();

    // 检查是否到达停止时间
    if (currentTime >= stopTime) {
        console.log("到达设定时间，程序已停止运行。");
        return; // 停止执行
    }

    // 获取屏幕尺寸
    const screenWidth = window.innerWidth || document.documentElement.clientWidth;
    const screenHeight = window.innerHeight || document.documentElement.clientHeight;

    // 计算屏幕中心
    const center_x = screenWidth / 2;
    const center_y = screenHeight / 2;

    // 定义点击的半径
    const radius = 80;

    // 生成中心区域内的随机位置
    const x = Math.floor(center_x + (Math.random() * 2 - 1) * radius);
    const y = Math.floor(center_y + (Math.random() * 2 - 1) * radius);

    // 创建并触发鼠标按下事件
    const mouseDownEvent = new MouseEvent("mousedown", {
        view: window,
        bubbles: true,
        cancelable: true,
        clientX: x,
        clientY: y,
        buttons: 1 // 鼠标左键
    });
    const targetElement = document.elementFromPoint(x, y);
    if (targetElement) {
        targetElement.dispatchEvent(mouseDownEvent);
    }

    // 模拟鼠标按钮按住的时间（50到200毫秒之间）
    const pressDuration = Math.random() * 150 + 50;
    setTimeout(() => {
        // 创建并触发鼠标释放事件
        const mouseUpEvent = new MouseEvent("mouseup", {
            view: window,
            bubbles: true,
            cancelable: true,
            clientX: x,
            clientY: y,
            buttons: 0 // 鼠标左键释放
        });
        targetElement.dispatchEvent(mouseUpEvent);

        // 创建并触发点击事件
        const clickEvent = new MouseEvent("click", {
            view: window,
            bubbles: true,
            cancelable: true,
            clientX: x,
            clientY: y
        });
        targetElement.dispatchEvent(clickEvent);
    }, pressDuration);

    // 生成随机延迟（1000到2000毫秒之间）
    const delay = 800 + Math.random() * 1000;

    // 输出调试信息
    console.log(`点击位置: (${x}, ${y}), 按住时间: ${pressDuration}ms, 下次点击延迟: ${delay.toFixed(2)}ms`);

    // 安排下一次点击
    setTimeout(randomCenterClick, delay + pressDuration);
}

// 开始随机点击循环
randomCenterClick();